.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,4 @@
+- [ ] Closes #xxxx (Replace xxxx with the GitHub issue number)
+- [ ] Tests added and passed (if fixing a bug or adding a new feature)
+- [ ] [Pre-commit code checks](https://pre-commit.com/#install) pass.
```

Comment:
I don't think this is necessary. Us maintainers can enforce it just by making the tests required before the merge.

Reply:
I wasn't thinking about enforcement, but rather efficiency.

Along the principle of [shift-left testing](https://en.m.wikipedia.org/wiki/Shift-left_testing), my thinking is that we can make the PR process more seamless by encouraging contributors to run the linter from the get-go. This saves contributors time (debugging failed Checks, making fixup commits), and maintainers time (on code review comments).

Reply:
Yup fair point.